Algorithm.CSharp/EmaCrossUniverseSelectionAlgorithm.cs

```diff
@@ -32,100 +32,116 @@ namespace QuantConnect.Algorithm.CSharp
+            foreach(var security in changes.RemovedSecurities)
+                Liquidate(security.Symbol, "Removed from Universe");
 
-            SetStartDate(2010, 01, 01);
-            SetEndDate(2015, 01, 01);
-            SetCash(100*1000);
-
-            AddUniverse(coarse =>
-            {
-                return (from cf in coarse
-                        // grab th SelectionData instance for this symbol
-                        let avg = _averages.GetOrAdd(cf.Symbol, sym => new SelectionData())
-                        // Update returns true when the indicators are ready, so don't accept until they are
-                        where avg.Update(cf.EndTime, cf.AdjustedPrice)
-                        // only pick symbols who have their 50 day ema over their 100 day ema
-                        where avg.Fast > avg.Slow*(1 + Tolerance)
-                        // prefer symbols with a larger delta by percentage between the two averages
-                        orderby avg.ScaledDelta descending
-                        // we only need to return the symbol and return 'Count' symbols
-                        select cf.Symbol).Take(Count);
-            });
+            canTrade = true;
         }
+        
 
         /// <summary>
         /// OnData event is the primary entry point for your algorithm. Each new data point will be pumped in here.
         /// </summary>
         /// <param name="data">TradeBars dictionary object keyed by symbol containing the stock data</param>
         public void OnData(TradeBars data)
         {
-            if (_changes == SecurityChanges.None) return;
-
-            // liquidate securities removed from our universe
-            foreach (var security in _changes.RemovedSecurities)
-            {
-                if (security.Invested)
-                {
-                    Liquidate(security.Symbol);
-                }
-            }
-
-            // we'll simply go long each security we added to the universe
-            foreach (var security in _changes.AddedSecurities)
-            {
-                SetHoldings(security.Symbol, TargetPercent);
+            if (canTrade) {
+                int length = symbols.Count;
+                
+                foreach(var symbol in symbols)
+                    SetHoldings(symbol, 1.0 / length);
+                
+                canTrade = false;
             }
         }
-
-        /// <summary>
-        /// Event fired each time the we add/remove securities from the data feed
-        /// </summary>
-        /// <param name="changes">Object containing AddedSecurities and RemovedSecurities</param>
-        public override void OnSecuritiesChanged(SecurityChanges changes)
+        
+        
+        public class SelectionData
         {
-            _changes = changes;
+            public ExponentialMovingAverage slow = new ExponentialMovingAverage(200);
+            public ExponentialMovingAverage fast = new ExponentialMovingAverage(50);
+            
```

Comment:
Read through [MS Naming Conventions](https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/naming-guidelines) and revert most of the variable name changes in the C# version.